### ISSUE TITLE
'gpg: keyserver receive failed: General error' when building image

### ISSUE THREAD

**Issue by schildbach** [cmehay/docker-tor-hidden-service#47]
When I try to build this image from source using

```
docker build --build-arg tor_version=0.4.3.6 .
```

I get

```
Sending build context to Docker daemon  557.1kB
Step 1/19 : FROM    python:3.7-alpine
 ---> 295b051ee125
Step 2/19 : ARG     tor_version
 ---> Using cache
 ---> 574908d63d94
Step 3/19 : ENV     HOME /var/lib/tor
 ---> Using cache
 ---> 392d96d0475e
Step 4/19 : ENV     POETRY_VIRTUALENVS_CREATE=false
 ---> Using cache
 ---> 8dda69c0fe7f
Step 5/19 : RUN     apk add --no-cache git bind-tools libevent-dev openssl-dev gnupg gcc make automake ca-certificates autoconf musl-dev coreutils libffi-dev zlib-dev &&     mkdir -p /usr/local/src/ /var/lib/tor/ &&     gpg --batch --auto-key-locate nodefault,wkd --recv-keys FE43009C4607B1FB &&     git clone https://git.torproject.org/tor.git /usr/local/src/tor &&     cd /usr/local/src/tor &&     git checkout tor-$tor_version &&     git verify-tag tor-$tor_version &&     ./autogen.sh &&     ./configure     --disable-asciidoc     --sysconfdir=/etc     --disable-unittests &&     make && make install &&     cd .. &&     rm -rf tor &&     pip3 install --upgrade pip poetry &&     apk del git libevent-dev openssl-dev gnupg make automake autoconf musl-dev coreutils libffi-dev &&     apk add --no-cache libevent openssl
 ---> Running in 984dfcf67cc2
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
(1/65) Installing m4 (1.4.18-r1)
(2/65) Installing perl (5.30.3-r0)
(3/65) Installing autoconf (2.69-r2)
(4/65) Installing automake (1.16.2-r0)
(5/65) Installing fstrm (0.6.0-r1)
(6/65) Installing libgcc (9.3.0-r2)
(7/65) Installing json-c (0.14-r1)
(8/65) Installing libstdc++ (9.3.0-r2)
(9/65) Installing libprotobuf (3.12.2-r0)
(10/65) Installing libprotoc (3.12.2-r0)
(11/65) Installing protobuf-c (1.3.3-r1)
(12/65) Installing libuv (1.38.1-r0)
(13/65) Installing libxml2 (2.9.10-r5)
(14/65) Installing bind-libs (9.16.6-r0)
(15/65) Installing bind-tools (9.16.6-r0)
(16/65) Installing libacl (2.2.53-r0)
(17/65) Installing libattr (2.4.48-r0)
(18/65) Installing coreutils (8.32-r0)
(19/65) Installing binutils (2.34-r1)
(20/65) Installing gmp (6.2.0-r0)
(21/65) Installing isl (0.18-r0)
(22/65) Installing libgomp (9.3.0-r2)
(23/65) Installing libatomic (9.3.0-r2)
(24/65) Installing libgphobos (9.3.0-r2)
(25/65) Installing mpfr4 (4.0.2-r4)
(26/65) Installing mpc1 (1.1.0-r1)
(27/65) Installing gcc (9.3.0-r2)
(28/65) Installing nghttp2-libs (1.41.0-r0)
(29/65) Installing libcurl (7.69.1-r1)
(30/65) Installing pcre2 (10.35-r0)
(31/65) Installing git (2.26.2-r0)
(32/65) Installing perl-error (0.17029-r0)
(33/65) Installing perl-git (2.26.2-r0)
(34/65) Installing git-perl (2.26.2-r0)
(35/65) Installing libgpg-error (1.37-r0)
(36/65) Installing libassuan (2.5.3-r0)
(37/65) Installing libcap (2.27-r0)
(38/65) Installing libblkid (2.35.2-r0)
(39/65) Installing libmount (2.35.2-r0)
(40/65) Installing pcre (8.44-r0)
(41/65) Installing glib (2.64.5-r0)
(42/65) Installing libgcrypt (1.8.5-r0)
(43/65) Installing libsecret (0.20.3-r0)
(44/65) Installing pinentry (1.1.0-r2)
Executing pinentry-1.1.0-r2.post-install
(45/65) Installing nettle (3.5.1-r1)
(46/65) Installing p11-kit (0.23.20-r5)
(47/65) Installing libtasn1 (4.16.0-r1)
(48/65) Installing libunistring (0.9.10-r0)
(49/65) Installing gnutls (3.6.15-r0)
(50/65) Installing libksba (1.4.0-r0)
(51/65) Installing db (5.3.28-r1)
(52/65) Installing libsasl (2.1.27-r6)
(53/65) Installing libldap (2.4.50-r0)
(54/65) Installing npth (1.6-r0)
(55/65) Installing gnupg (2.2.23-r0)
(56/65) Installing python3 (3.8.5-r0)
(57/65) Installing libevent (2.1.11-r1)
(58/65) Installing pkgconf (1.7.2-r0)
(59/65) Installing libevent-dev (2.1.11-r1)
(60/65) Installing linux-headers (5.4.5-r1)
(61/65) Installing libffi-dev (3.3-r2)
(62/65) Installing make (4.3-r0)
(63/65) Installing musl-dev (1.1.24-r9)
(64/65) Installing openssl-dev (1.1.1g-r0)
(65/65) Installing zlib-dev (1.2.11-r3)
Executing busybox-1.31.1-r16.trigger
OK: 285 MiB in 100 packages
gpg: directory '/var/lib/tor/.gnupg' created
gpg: keybox '/var/lib/tor/.gnupg/pubring.kbx' created
gpg: keyserver receive failed: General error
The command '/bin/sh -c apk add --no-cache git bind-tools libevent-dev openssl-dev gnupg gcc make automake ca-certificates autoconf musl-dev coreutils libffi-dev zlib-dev &&     mkdir -p /usr/local/src/ /var/lib/tor/ &&     gpg --batch --auto-key-locate nodefault,wkd --recv-keys FE43009C4607B1FB &&     git clone https://git.torproject.org/tor.git /usr/local/src/tor &&     cd /usr/local/src/tor &&     git checkout tor-$tor_version &&     git verify-tag tor-$tor_version &&     ./autogen.sh &&     ./configure     --disable-asciidoc     --sysconfdir=/etc     --disable-unittests &&     make && make install &&     cd .. &&     rm -rf tor &&     pip3 install --upgrade pip poetry &&     apk del git libevent-dev openssl-dev gnupg make automake autoconf musl-dev coreutils libffi-dev &&     apk add --no-cache libevent openssl' returned a non-zero code: 2
```

**Comment by schildbach** [comment]
I'm not entirely sure, but it seems that the key FE43009C4607B1FB which this build relies on expired on 2019-09-21.

**Comment by cmehay** [comment]
Yes this is the issue.

As the current stable tag was signed with this expired key, the only workaround is to accept signature with an expired key... or wait for the next release to update it...

**Comment by schildbach** [comment]
I think we should accept signatures with an expired key in the Dockerfile. Builds are supposed to succeed even in years, a time-bomb is inappropriate here imho.

I'm happy to provide a PR if you agree.

**Comment by cmehay** [comment]
yes please do

**Comment by NoNamePro0** [comment]
Any updates, relating this issue?

**Comment by cmehay** [comment]
I don't find any satisfying workaround so I'm removing the signature check.

It should be good as long the repository is cloned from https://git.torproject.org/tor.git which is already authenticated by TLS.

**Comment by NoNamePro0** [comment]
hmm, okay. Maybe do that with [git submodules.](https://git-scm.com/book/en/v2/Git-Tools-Submodules)

**Comment by schildbach** [comment]
I'm closing this as the issue has been resolved via b7bf5f61b06f0bc1920585e0379aebe8587066db.

**Comment by NoNamePro0** [comment]
Great!
